src/lib/shapelib/library.ts

```diff
@@ -64,7 +64,26 @@ export interface LoopyLineParams {
 }
 
 export function loopyLine(params: LoopyLineParams): PathArray {
-  // TODO: make loopy line procedural
+  /*
+
+    TODO: Make loopy line procedural.
```

Comment:
@vincentropy See here.

Reply:
+1

There is already some rudimentary support for composing shapes, but it makes sense to add some meta data that specify connection points to allow more advanced composition. 

I'll document some of this better in the readme